packages/back-end/src/models/DataSourceModel.ts

```diff
@@ -162,10 +164,11 @@ export async function createDataSource(
     projects,
   };
 
-  // Test the connection and create in the database
   await testDataSourceConnection(datasource);
   const model = await DataSourceModel.create(datasource);
 
+  await onDataSourceCreate(datasource, organization);
```

Comment:
I think it makes sense to queue this as a background job with Agenda instead of awaiting it here.  Especially if it's a very large database, this could easily take 30+ seconds to run and we don't want the "Create" button to stay in a loading state that long.